File: bench/dbicpp-test.cc
#include "dbic++.h"
#include <cstdio>

using namespace std;
using namespace dbi;

int main(int argc, char *argv[]) {
    int max;
    dbiInitialize("../libs");
    max = argc == 2 ? atoi(argv[1]) : 10000;
    Handle h = dbiConnect("postgresql", "udbicpp", "", "dbicpp");

    Statement st(h, "SELECT id, name, email FROM users");
    ResultRow r;

    for (int n = 0; n < max; n++) {
        st.execute();
        while ((r = st.fetchRow()).size() > 0)
            printf("%s\t%s\t%s\n", r[0].value.c_str(), r[1].value.c_str(), r[2].value.c_str());
        st.finish();
    }
}

File: bench/dbicpp-test.rb
#!/usr/bin/ruby

require_relative '../ruby/dbicpp'

DBI.init File.dirname(__FILE__) + '/../libs'

h = DBI::Handle.new user: 'udbicpp', db: 'dbicpp', driver: 'postgresql'

st = h.prepare "SELECT * FROM users"

(ARGV[0] || 25000).to_i.times {
  st.execute
  while r = st.fetchrow
    p r
  end
}
